test/FlowChartSpec.hs

```diff
@@ -0,0 +1,84 @@
+{-# LANGUAGE QuasiQuotes #-}
+
+module FlowChartSpec
+  ( spec,
+  )
+where
+
+import Algebra.Graph.Labelled as LG (edgeList, vertexList)
+import FlowChart
+import NeatInterpolation
+import Test.Hspec
+import Text.Megaparsec
+
+shouldHaveEdges :: Maybe Diagram -> [(Maybe Edge, Node, Node)] -> Expectation
+shouldHaveEdges = \x -> (\y -> shouldBe (edgeList . graph <$> x) (Just y))
```

Comment:
Since you are always passing `Just ...`, the second argument should use type `Edge` rather than `Maybe`. 

Also, there is no particular reason to use a lambda here? When in doubt, use normal syntax:

```haskell
shouldHaveEdges diagram edges = 
  fmap (edgeList . graph) diagram `shouldBe` Just edges
```

This ^ reads more naturally.

Generally speaking, it is a good idea to *refactor* a function after you have written it. Take a bird's eye view, and imagine that someone else is reading trying to understand it just by glancing.